device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/fileupload/FileUploadTask.java

```diff
@@ -199,9 +208,31 @@ public IotHubStatusCode sendNotification(FileUploadCompletionNotification fileUp
         ResponseMessage responseMessage = httpsTransportManager.sendFileUploadNotification(message);
         httpsTransportManager.close();
 
+        validateServiceStatusCode(responseMessage, "Failed to complete the file upload notification");
+
         return responseMessage.getStatus();
     }
 
+    private void validateServiceStatusCode(ResponseMessage responseMessage, String errorMessage) throws IOException
```

Comment:
why not have this method return the responseMessagePayload so that it doesn't need to be calculated again in L188?

Reply:
Sure, that seems reasonable